environments/common/src/main/java/org/jboss/weld/environment/deployment/discovery/jandex/JandexDiscoveryStrategy.java

```diff
@@ -90,8 +90,16 @@ protected WeldBeanDeploymentArchive processAnnotatedDiscovery(BeanArchiveBuilder
         while (classIterator.hasNext()) {
             String className = classIterator.next();
             ClassInfo cinfo = cindex.getClassByName(DotName.createSimple(className));
-            if (!containsBeanDefiningAnnotation(cinfo, className)) {
-                classIterator.remove();
+            if (cinfo != null) {
+                if (!containsBeanDefiningAnnotation(cinfo, className)) {
+                    classIterator.remove();
+                }
+            } else {
+                //if ClassInfo is not available (e.g for WEB-INF/lib/jars) then fallback to reflection
+                Class<?> clazz = Reflections.loadClass(resourceLoader, className);
+                if (!Reflections.hasBeanDefiningAnnotation(clazz, initialBeanDefiningAnnotations)) {
```

Comment:
Then it would make more sense to define it as:
`clazz == null || !Reflections.hasBeanDefiningAnnotation(clazz, initialBeanDefiningAnnotations)`
or not?